### DISCUSSION TITLE
How to improve stockfish?

### DISCUSSION THREAD

**Discussion by steve01101** [official-stockfish/Stockfish#4601] [General]
Hi everyone, I've been in the chess world for quite a while and the continuous improvement of chess fascinates me. I was thinking how can Stockfish be improved? I would like to try to discuss it with you. Maybe stockfish is improving every day, maybe it's not so easy to talk about improvement, but I was curious to know if something already excellent could be improved. I would like to know more about stockfish if there are wikis that I have to read I'll go right away, I've read something about it it's very interesting but at the same time complex. I was thinking, and you will tell me, maybe the code can be improved? Make it lighter? Can more efficient algorithms be implemented to reach greater depths? From what I understand with the NNUE technology and the ability to play stockfish, does he get better and better? I don't know if the Monte-Carlo Tree Search can be implemented, or if it is already implemented. I would like to know more. I don't know if you can implement some hardware acceleration, make move choices faster, calculate only the most probable moves. I like this whole world.

**Comment by cj5716** [comment]
It is constantly improving on fishtest, https://tests.stockfishchess.org/tests
https://github.com/glinscott/fishtest/wiki/Creating-my-first-test is a helpful guide for you.
You can also learn about various techniques used in chess engines here https://www.chessprogramming.org/Main_Page
Join our friendly discord server https://discord.com/invite/GWDRS3kU6R for support/discussions
